test/e2e/hot-reload.test.js

```diff
@@ -145,14 +140,15 @@ describe('hot-reload', () => {
 
       const watcher = chokidar.watch(testGatewayConfigPath, watchOptions);
       watcher.once('change', (evt) => {
-        request
-          .get(`http://localhost:${originalGatewayPort}`)
-          .end((err, res) => {
-            assert(err);
-            assert(res.clientError);
-            assert.equal(res.statusCode, 404);
-            done();
-          });
+        setTimeout(() => {
+          request
+            .get(`http://localhost:${originalGatewayPort}`)
+            .end((err, res) => {
+              assert(res.clientError);
+              assert.equal(res.statusCode, 404);
+              done(err);
+            });
+        }, 5000);
```

Comment:
what if resubscribe to on change again instead of timeout?

Reply:
Oh - I'm still working on this - I'll ping you once done.